KeepSafe/aiohttp
# Added ServiceRestart and TryAgainLater websocket close codes

## What these changes does?

Allows using IANA standard websocket close codes `1012 Service Restart` and `1013 Try Again Later`

## How to test your changes?

```python
async def myws(request):
    ws = WebSocketResponse()
    await ws.prepare(request)
    # ...
    await ws.close(code=1012, message='Service Restart')
```
Expected behaviour: websocket closed with 1012 code
Actual behaviour without this PR: 
```
[2016-03-14 00:05:08 +0300] [26962] [ERROR] Error handling request
Traceback (most recent call last):
  File "/usr/lib/python3.5/site-packages/aiohttp/web_ws.py", line 230, in receive
    msg = yield from self._reader.read()
  File "/usr/lib/python3.5/site-packages/aiohttp/streams.py", line 591, in read
    result = yield from super().read()
  File "/usr/lib/python3.5/site-packages/aiohttp/streams.py", line 446, in read
    yield from self._waiter
  File "/usr/lib/python3.5/asyncio/futures.py", line 358, in __iter__
    yield self  # This tells Task to wait for completion.
  File "/usr/lib/python3.5/asyncio/tasks.py", line 290, in _wakeup
    future.result()
  File "/usr/lib/python3.5/asyncio/futures.py", line 274, in result
    raise self._exception
  File "/usr/lib/python3.5/site-packages/aiohttp/parsers.py", line 139, in feed_data
    self._parser.send(data)
  File "/usr/lib/python3.5/site-packages/aiohttp/websocket.py", line 86, in WebSocketParser
    'Invalid close code: {}'.format(close_code))
aiohttp.websocket.WebSocketError: Invalid close code: 1012
```

## Related issue number

None

## Checklist

- [x] Code is written and well
- [x] Tests for the changes are provided

These codes are listed in IANA registry: https://www.iana.org/assignments/websocket/websocket.xml#table-close-code-number
But without this change `aiohttp` doesn't allow to use them.
